pkg/controller/signup.go

```diff
@@ -36,6 +37,11 @@ func NewSignup(app application.Application, config configuration.Configuration)
 // PostHandler creates a Signup resource
 func (s *Signup) PostHandler(ctx *gin.Context) {
 	userSignup, err := s.app.SignupService().Signup(ctx)
+	if err, ok := err.(*errors2.StatusError); ok {
+		errors.AbortWithError(ctx, int(err.Status().Code), err, "error creating UserSignup resource")
```

Comment:
Could we maybe create a new function similar to `errors.AbortWithError()` which would accept an error object and either use the status code if it's StatusError (as you do here) or use 500 by default.
Then we could either explicitly set the expected code when we know what we want (eg. `errors.AbortWithError(ctx, code, err, message)`) or do something like `errors.AbortWithStatusError(ctx, err, message)` when we want to use the code from the corresponding StatusError or use 500 by default.

Reply:
@alexeykazakov Yup, I think that's possible and sounds like a good idea. We can maybe add this to the reg-service errors package. There is already an AbortWithError there that we can re-purpose to do as you described above

Reply:
I'm not sure if we can re-purpose `AbortWithError()` because we need to specify the error code explicitly in some cases. It seems that we need both functions. So, I would keep `AbortWithError()` as is but also introduce a new function where we delegate the error code setting. Something like `AbortWithStatusError()`. See an example in my previous comment.